aws/delta-checkpoint/src/main.rs

```diff
@@ -0,0 +1,201 @@
+//! Lambda function for writing checkpoints from S3 event triggers on _delta_log JSON files.
+//!
+
+// Reference:
+//
+// https://aws.amazon.com/blogs/opensource/rust-runtime-for-aws-lambda
+// https://docs.aws.amazon.com/cli/latest/reference/s3api/put-bucket-notification-configuration.html
+// https://docs.aws.amazon.com/lambda/latest/dg/with-s3.html
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-how-to-filtering.html
+
+use deltalake::checkpoints::{CheckPointWriter, CheckPointWriterError};
+use deltalake::DeltaDataTypeVersion;
+use lambda_runtime::{handler_fn, Context, Error};
+use lazy_static::lazy_static;
+use log::*;
+use regex::Regex;
+use serde_json::Value;
+use std::num::ParseIntError;
+use std::path::PathBuf;
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let func = handler_fn(func);
+    lambda_runtime::run(func).await?;
+    Ok(())
+}
+
+async fn func(event: Value, _: Context) -> Result<(), CheckPointLambdaError> {
+    process_event(&event).await
+}
+
+async fn process_event(event: &Value) -> Result<(), CheckPointLambdaError> {
```

Comment:
Yeah, looking at the event structure, there are a ton of fields that I don't need. I figured just plucking the JSON out would be faster to avoid the deserialization. I'm happy to change this if you think its advisable.